govcd/nsxt_edgegateway.go

```diff
@@ -0,0 +1,302 @@
+}
+
+// GetNsxtEdgeGatewayByName allows to retrieve NSX-T edge gateway by Name for specifi Vdc
+func (vdc *Vdc) GetNsxtEdgeGatewayByName(name string) (*NsxtEdgeGateway, error) {
+	queryParameters := url.Values{}
+	queryParameters.Add("filter", "name=="+name)
+
+	allEdges, err := vdc.GetAllNsxtEdgeGateways(queryParameters)
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve edge gateway by name '%s': %s", name, err)
+	}
+
+	onlyNsxtEdges := filterOnlyNsxtEdges(allEdges)
+
+	return returnSingleNsxtEdgeGateway(name, onlyNsxtEdges)
+}
+
+// GetAllNsxtEdgeGateways allows to retrieve all NSX-T edge gateways for Org Admins
+func (adminOrg *AdminOrg) GetAllNsxtEdgeGateways(queryParameters url.Values) ([]*NsxtEdgeGateway, error) {
+	return getAllNsxtEdgeGateways(adminOrg.client, queryParameters)
+}
+
+// GetAllNsxtEdgeGateways  allows to retrieve all NSX-T edge gateways for Org users
+func (org *Org) GetAllNsxtEdgeGateways(queryParameters url.Values) ([]*NsxtEdgeGateway, error) {
+	return getAllNsxtEdgeGateways(org.client, queryParameters)
+}
+
+// GetAllNsxtEdgeGateways  allows to retrieve all NSX-T edge gateways for Org users
+func (vdc *Vdc) GetAllNsxtEdgeGateways(queryParameters url.Values) ([]*NsxtEdgeGateway, error) {
+	filteredQueryParams := queryParameterFilterAnd("orgVdc.id=="+vdc.Vdc.ID, queryParameters)
+	return getAllNsxtEdgeGateways(vdc.client, filteredQueryParams)
+}
+
+// CreateNsxtEdgeGateway allows to create NSX-T edge gateway for Org admins
+func (adminOrg *AdminOrg) CreateNsxtEdgeGateway(edgeGatewayConfig *types.OpenAPIEdgeGateway) (*NsxtEdgeGateway, error) {
+	if !adminOrg.client.IsSysAdmin {
+		return nil, fmt.Errorf("only System Administrator can create Edge Gateway")
+	}
+
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointEdgeGateways
+	minimumApiVersion, err := adminOrg.client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := adminOrg.client.OpenApiBuildEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	returnEgw := &NsxtEdgeGateway{
+		EdgeGateway: &types.OpenAPIEdgeGateway{},
+		client:      adminOrg.client,
+	}
+
+	err = adminOrg.client.OpenApiPostItem(minimumApiVersion, urlRef, nil, edgeGatewayConfig, returnEgw.EdgeGateway)
+	if err != nil {
+		return nil, fmt.Errorf("error creating Edge Gateway: %s", err)
+	}
+
+	return returnEgw, nil
+}
+
+// Update allows to update NSX-T edge gateway for Org admins
+func (egw *NsxtEdgeGateway) Update(edgeGatewayConfig *types.OpenAPIEdgeGateway) (*NsxtEdgeGateway, error) {
+	if !egw.client.IsSysAdmin {
+		return nil, fmt.Errorf("only System Administrator can update Edge Gateway")
+	}
+
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointEdgeGateways
+	minimumApiVersion, err := egw.client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	if edgeGatewayConfig.ID == "" {
+		return nil, fmt.Errorf("cannot update Edge Gateway without id")
```

Comment:
Two nits/suggestions:

1. What about using capital `ID` in user facing messages?
2. I really like when you upper case `Edge Gateway` in the messages, but this PR has other places where it's just `edge gateway`. Would be great to be consistent.

Reply:
This is totally a place where I usually fall on following consistency. Grep should have done it.